scripts/documentation/typedoc_parser.ts

```diff
@@ -0,0 +1,237 @@
+     *
+     * Examples:
+     * TypedocParser.paramsString(params);
+     * => "agreementId: string"
+     *
+     * @param {SignatureParameter[]} params
+     * @returns {string}
+     */
+    private static paramsString(params: SignatureParameter[]) {
+        return _.map(params, (param) => {
+            return `${param.name}: ${param.type.name}`;
+        }).join(",<br/>");
+    }
+
+    private static sectionName(classObj): string {
+        // E.g. "adapters" from Typedoc name "adapters/simple_interest_loan_adapter".
+        return classObj.name.split("/")[0].substr(1);
+    }
+
+    private static getClassMethods(classObj) {
+        return _.filter(classObj.children[0].children, (child) => {
+            return child.kindString === "Method";
+        });
+    }
+
+    private static isPromise(signature): boolean {
+        return signature.type === "Promise" &&
+            signature.type.typeArguments &&
+            signature.type.typeArguments.length;
+    }
+
+    private static returnType(signature) {
+        const type = signature.type.name;
+
+        const isPromise = TypedocParser.isPromise(signature);
+
+        const promiseTarget = isPromise ? ("<" + signature.type.typeArguments[0].name + ">") : "";
+
+        return type + promiseTarget;
+    }
+
+    // The path to the Typedoc input JSON file.
+    private readonly filePath;
+    // The typedoc JSON input, as read from the JSON file at `filePath`.
+    private input: TypedocInput;
+    // The documentation output, based on the Typedoc data.
+    private output: Documentation;
+
+    constructor(filePath: string) {
+        this.filePath = filePath;
+    }
+
+    public parse(): void {
+        // Read the contents of the Typedoc file.
+        this.readFile();
+        // Transform the Typedoc JSON into human-friendly output.
+        this.parseData();
+    }
+
+    /**
+     * Given a the desired output path, saves a file there containing
+     * the human-friendly JSON data.
+     *
+     * @param {string} outputPath
+     */
+    public writeToFile(outputPath: string): void {
+        // Convert the documentation object into a string, so we can write it to a file.
+        const contents = JSON.stringify(this.output);
+
+        fs.writeFileSync(outputPath, contents);
+    }
+
+    private parseData(): void {
+        this.output = { sections: this.getSections() };
+    }
+
+    private getSections(): SectionDocumentation[] {
+        const groupedClasses = this.classesPerSection();
+
+        return _.map(groupedClasses, this.getSectionData.bind(this));
+    }
+
+    private getSectionData(classes: object[], groupName: string): SectionDocumentation {
+        return {
+            title: groupName,
+            classes: this.getClassData(classes),
+        };
+    }
+
+    private classesPerSection(): SectionToTypedocClasses {
+        const allClasses = this.getClasses();
+
+        return _.groupBy(allClasses, TypedocParser.sectionName);
```

Comment:
From the lodash docs, looks like `groupBy` determines the ordering of the grouped values by "the order they occur in the collection."  Do we have any guarantees about ordering from typedoc?  I imagine we'd want these to be sorted alphabetically?